staging/src/k8s.io/apimachinery/pkg/util/validation/field/errors.go

```diff
@@ -200,12 +200,12 @@ func Invalid(field *Path, value interface{}, detail string) *Error {
 // NotSupported returns a *Error indicating "unsupported value".
 // This is used to report unknown values for enumerated fields (e.g. a list of
 // valid values).
-func NotSupported(field *Path, value interface{}, validValues []string) *Error {
+func NotSupported[S ~string](field *Path, value interface{}, validValues []S) *Error {
```

Comment:
I didn't expect to change the apimachinery library function in a way that required callers to react like this

Reply:
maybe I'm misreading this... were callers required to change or not?

Reply:
> The question is whether users of NotSupported that are currently working had to do any changes.

Ah, I see. Thank you for clarifying!

Reply:
@alculquicondor As you say, this change doesn't have backward compatibility. So, working on a separate PR might be better. 

WDYT?

Reply:
good from my side